include/FrictionQPotSpringBlock/Line1d.hpp

```diff
@@ -916,6 +916,65 @@ inline size_t System::triggerWeakest(double eps, int direction)
     return p;
 }
 
+////////////////////System_thermal begins////////////////////////////////
+template <class T>
+inline System_thermal::System_thermal(
+    double tmp,
+    double m,
+    double eta,
+    double mu,
+    double k_neighbours,
+    double k_frame,
+    double dt,
+    const T& x_y){
+        xt::xtensor<long, 1> istart = xt::zeros<long>({x_yield.shape(0)});
+        m_model.init(m, eta, mu, k_neighbours, k_frame, dt, x_y, istart);
+        m_thermal = tmp;
+        m_teprt = 0;
+        m_N = x_y.shape(0);
+        m_f_thermal = xt::zeros<double>({m_N});
+}
+
+template <class T, class I>
+inline System_thermal::System_thermal(
+    double tmp,
+    double m,
+    double eta,
+    double mu,
+    double k_neighbours,
+    double k_frame,
+    double dt,
+    const T& x_y,
+    const I& istart){
+        m_model.init(m, eta, mu, k_neighbours, k_frame, dt, x_y, istart);
+        m_thermal = tmp;
+        m_teprt = 0;
+        m_N = x_y.shape(0);
+        m_f_thermal = xt::zeros<double>({m_N});
+}
+
+inline void System_thermal::timeStep(){
+    m_model.timeStep();
+    this->ComputeForceThermally();
+    xt::noalias(m_model.m_a) = m_model.m_f / m_model.m_m;
```

Comment:
Here things go wrong in the integrator. That is why I would urge you derive `System_thermal` (or maybe `SystemThermalRandomForcing`) from `System` and `override `computeForce`. The problem is that the integration scheme needs to evaluate the residual forces twice to come up with a good estimate of `m_a`. Currently, this first time the system would not feel the thermal forces, which would lead to spurious behaviour.